src/lib/tracking-consensus.js

```diff
@@ -0,0 +1,70 @@
+class TrackingConsensus {
+    static init() {
+        const settings = this._readStorage();
+        if (typeof settings.allowsBrowserData === 'undefined') {
+            // Not yet set, ask user
+            document.getElementById('tracking-consensus').classList.remove('display-none');
+        } else {
+            if (this.allowsTracking(settings)) this._appendScript();
+        }
+    }
+
+    static allowsBrowserData() {
+        return this._readStorage().allowsBrowserData;
+    }
+
+    static allowsUsageData() {
+        return this._readStorage().allowsUsageData;
+    }
+
+    static allowsTracking(settings) {
+        console.debug("Tracking settings:", settings);
+        return settings.allowsBrowserData || settings.allowsUsageData;
+    }
+
+    /**
+     * @params {{allowsBrowserData?: boolean, allowsUsageData?: boolean}} obj
+     */
+    static update(obj) {
+        document.getElementById('tracking-consensus').classList.add('display-none');
+
+        const settings = Object.assign(this._readStorage(), obj);
+        localStorage.setItem(TrackingConsensus.STORAGE_KEY, JSON.stringify(settings));
+
+        // If at least one of the settings is true, start the tracking script
+        if (this.allowsTracking(settings)) this._appendScript();
+    }
+
+    static initPaq() {
+        window._paq = [];
+        _paq.push(['trackPageView']);
+        _paq.push(['enableLinkTracking']);
+    }
+
+    static _readStorage() {
+        const storage = localStorage.getItem(TrackingConsensus.STORAGE_KEY);
```

Comment:
Not critical, just a remark.

This could make use of a similar mechanism to appendScript to reduce read operations on localStorage. I don't think they are critical in terms of access performance though.

Could also be a Singleton to only ever read once, too.